simpeg_drivers/plate_simulation/sweep/options.py

```diff
@@ -0,0 +1,157 @@
+
+
+class SweepOptions(Options):
+    """
+    Options for sweeping parameters within a template application.
+
+    :param template: A SimPEGGroup containing the template for running an application.
+        Any unswept parameters required by the application must be set on the groups
+        options.  Any swept parameters will take priority over those set on the groups
+        options.
+    :param sweeps:  Sweep parameters to be combined to create a series of trials run
+        by the template application.
+    """
+
+    name: ClassVar[str] = "plate_sweep"
+    default_ui_json: ClassVar[Path] = assets_path() / "uijson/plate_sweep.ui.json"
+    title: ClassVar[str] = "Plate Sweep"
+    run_command: ClassVar[str] = "simpeg_drivers.plate_simulation.sweep.driver"
+    out_group: SimPEGGroup | None = None
+    template: SimPEGGroup | UIJsonGroup
+    sweeps: list[ParamSweep]
+    workdir: Path | None = None
+
+    @field_serializer("sweeps")
+    def sweeps_to_params(self, sweeps):
+        out = {}
+        for sweep in sweeps:
+            if sweep.stop is not None and sweep.count < 2:
+                out[f"{sweep.name}_start"] = sweep.start
+                continue
+            for key, value in sweep.model_dump().items():
+                if key == "name":
+                    continue
+                out[f"{sweep.name}_{key}"] = value
+        return out
+
+    @field_serializer("workdir")
+    def workdir_to_string(self, workdir):
+        return str(workdir)
+
+    @classmethod
+    def build(cls, input_data: InputFile | dict | None = None, **kwargs) -> Self:
+        """
+        Build a dataclass from a dictionary or InputFile.
+
+        :param input_data: Dictionary of parameters and values.
+
+        :return: Dataclass of application parameters.
+        """
+        data = input_data or {}
+        if isinstance(input_data, InputFile) and input_data.data is not None:
+            data = input_data.data.copy()
+
+        if not isinstance(data, dict):
+            raise TypeError("Input data must be a dictionary or InputFile.")
+
+        data.update(kwargs)
+        options = Options.collect_input_from_dict(cls, data)  # type: ignore
+
+        def collect_sweep(param: str) -> dict:
+            return {
+                "name": param,
+                "start": options.get(f"{param}_start"),
+                "stop": options.get(f"{param}_stop"),
+                "count": options.get(f"{param}_count"),
+            }
+
+        sweep_params = [k.removesuffix("_start") for k in options if "_start" in k]
+        options["sweeps"] = [collect_sweep(param) for param in sweep_params]
+        workdir = options["workdir"]
+        if isinstance(workdir, str):
+            options["workdir"] = Path(workdir)
+        if isinstance(workdir, list):
+            if len(workdir) == 0:
+                options["workdir"] = None
+            else:
+                options["workdir"] = Path(workdir[0])
```

Comment:
```suggestion
        options = Options.collect_input_from_dict(cls, data)  # type: ignore
```
Can we overload the `collect_input_from_dict` on this class to re-arrange the sweep components? You can call super().collect_input_from_dict first in there.

That way we don't need to replicate the mechanics of `start` here

Reply:
Yes, I was able to rework this, but with a geoapps-utils PR.